Tasks/SonarQubePostTest/PRCA/PostComments-Module.psm1

```diff
@@ -50,289 +52,195 @@ $script:project = $null
-    
-    $baseVersionDescriptor = New-Object -TypeName "Microsoft.TeamFoundation.SourceControl.WebApi.GitBaseVersionDescriptor"
-    $baseVersionDescriptor.VersionType = [Microsoft.TeamFoundation.SourceControl.WebApi.GitVersionType]::Commit
-    $baseVersionDescriptor.Version = $script:pullRequest.LastMergeTargetCommit.CommitId
-    
-    $commitDiffs = $script:gitClient.GetCommitDiffsAsync(
-        $script:project, # string project 
-        $script:pullRequest.Repository.Name, # string repositoryId 
-        $true, # bool? diffCommonCommit
-        $null, # int? top 
-        $null, # int? skip
-        $baseVersionDescriptor, 
-        $targetVersionDescriptor, 
-        $null, # object userState
-        [System.Threading.CancellationToken]::None # CancellationToken cancellationToken
-        ).Result
-    
-    if ($commitDiffs.ChangeCounts.Count -gt 0)
+    param ([ValidateNotNull()][Array]$messages)
+    
+    foreach ($message in $messages)
     {
-        Write-Verbose "Found $($commitDiffs.ChangeCounts.Count) changed file(s) in the PR"
-        
-        $sourceFiles = $commitDiffs.Changes | 
-            Where-Object { ($_ -ne $null) -and ($_.Item.IsFolder -eq $false) }  | 
-            ForEach-Object { $_.Item.Path.Replace("\", "/") }
+        Assert (![String]::IsNullOrEmpty($message.RelativePath)) "A message doesn't have a RelativePath property"
+        Assert (![String]::IsNullOrEmpty($message.Priority)) "A message doesn't have a Priority property"
+        Assert (![String]::IsNullOrEmpty($message.Content)) "A message doesn't have content "
     }
-
-    return $sourceFiles
 }
 
-#
-# Posts the discussion threads loaded with comments to the PR
-#
-function PostDiscussionThreads
+function InternalPostAndResolveComments
 {
-    param ([ValidateNotNull()][Microsoft.VisualStudio.Services.CodeReview.Discussion.WebApi.DiscussionThreadCollection]$threads)
+    param ([ValidateNotNull()][Array]$messages, [string][ValidateNotNullOrEmpty()]$messageSource)
     
-    $vssJsonThreadCollection = New-Object -TypeName "Microsoft.VisualStudio.Services.WebApi.VssJsonCollectionWrapper[Microsoft.VisualStudio.Services.CodeReview.Discussion.WebApi.DiscussionThreadCollection]" -ArgumentList @(,$threads)
-    [void]$script:discussionClient.CreateThreadsAsync($vssJsonThreadCollection, $null, [System.Threading.CancellationToken]::None).Result
+    Write-Verbose "Fetching existing threads and comments..."
     
-    Write-Host "Posted $($threads.Count) discussion threads"
-}
-
+    # TODO: are these needed here or can the build dictionary fetch them? 
+    $existingThreads = FetchActiveDiscussionThreads 
+    $existingComments = FetchDiscussionComments $existingThreads    
 
-#
-# Retrieve existing discussion threads. 
-#
-function FetchDiscussionThreads
-{
-    $threadsDictionary = $script:discussionClient.GetThreadsAsync( @($script:artifactUri)).Result
+    BuildMessageToCommentDictonary $messages $existingThreads $existingComments
```

Comment:
Instead of going through each thread, comment and message, I decided to keep a dictionary of message -> list of comments between the resolve comment phase and post new comments phase.